### PR TITLE
CONCF-504 Trending articles

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-504

https://github.com/Wikia/mercury/pull/764 was targeted at an obsolete branch.

### CODE REVIEW

**Comment by rogatty** [comment]
LGTM

File: front/scripts/main/components/TrendingArticlesItemComponent.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			this.lazyLoadImage();
+		}
+	},
+
+	viewportObserver: Em.observer('viewportDimensions.width', function (): void {
+		this.updateImageSize(this.get('viewportDimensions.width'));
+	}),
+
+	lazyLoadImage: function (): void {
+		var options: any = {},
```
Comment (hakubo) [inline front/scripts/main/components/TrendingArticlesItemComponent.ts]: can we initialize with declaration?
```
options = {
  width = this.get('imageWidth'),
  height = this.get('imageHeight'),
  mode = this.get('cropMode')
}
```

File: front/scripts/main/components/TrendingArticlesItemComponent.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+		options.width = this.get('imageWidth');
+		options.height = this.get('imageHeight');
+		options.mode = this.get('cropMode');
+		imageUrl = this.thumbnailer.getThumbURL(this.get('imageUrl'), options);
+		this.set('imageUrl', imageUrl);
+	},
+
+	updateImageSize: function (viewportWidth: number): void {
+		var componentWidth = String((viewportWidth - 20) / 2);
```
Comment (hakubo) [inline front/scripts/main/components/TrendingArticlesItemComponent.ts]: what is this 20?
what is this 2?

File: front/styles/main/component/_trending-articles.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	// this pseudoelement lets the last inline item to be justified to the right if it's even
+	&::after {
+		content: '';
+		display: inline-block;
+		width: 100%;
+	}
+
+	h4 {
+		color: $color-blue-light;
+		font-size: 1.25rem;
```
Comment (hakubo) [inline front/styles/main/component/_trending-articles.scss]: I think we have font size defined in vars or settings

File: front/scripts/main/components/TrendingArticlesItemComponent.ts (hunk 1 above)
Comment (Warkot) [inline front/scripts/main/components/TrendingArticlesItemComponent.ts]: An what do you think on initializing `imageUrl` this way then?
```
var options: any = {
		width: this.get('imageWidth'),
		height: this.get('imageHeight'),
		mode: this.get('cropMode')
	},
	imageUrl: string = this.thumbnailer.getThumbURL(this.get('thumbnail'), options);
```

Comment (rogatty) [inline front/scripts/main/components/TrendingArticlesItemComponent.ts]: No need for `: any`, other than that it's fine :+1:

**Comment by rogatty** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

File: front/scripts/main/components/CuratedContentItemComponent.ts [hunk 4]
```diff
@@ -39,7 +39,7 @@ App.CuratedContentItemComponent = Em.Component.extend(App.ViewportMixin, {
 		this.set('thumbUrl', thumbUrl);
 	},
 
-	updateImageSize: function(viewportSize: number) {
+	updateImageSize: function(viewportSize: number): void {
```
Comment (kenkouot) [inline front/scripts/main/components/CuratedContentItemComponent.ts]: space after function keyword or use object method shorthand

File: front/scripts/main/components/TrendingArticlesItemComponent.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	tagName: 'a',
+	classNames: ['trending-articles-item'],
+	attributeBindings: ['href', 'style'],
+	cropMode: Mercury.Modules.Thumbnailer.mode.topCrop,
+	thumbnailer: Mercury.Modules.Thumbnailer,
+	emptyGif: 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAQAIBRAA7',
+	currentlyRenderedImageUrl: Em.computed.oneWay('emptyGif'),
+	href: Em.computed.oneWay('url'),
+	imageWidth: 250,
+	imageHeight: Em.computed(function (): number { return Math.floor(this.get('imageWidth') * 9 / 16); }),
```
Comment (kenkouot) [inline front/scripts/main/components/TrendingArticlesItemComponent.ts]: lets not one line this, even if it is small

File: front/scripts/main/components/CuratedContentItemComponent.ts (hunk 4 above)
Comment (Warkot) [inline front/scripts/main/components/CuratedContentItemComponent.ts]: Do we have any pre-commit hooks defined for Mercury like we do for app repo? Any separate mercury-guidelines repo? Or would it be effective to run ones from guidelines repo?

**Pull request merged**